### ISSUE TITLE
Custom data training has constant loss regardless epoch

### ISSUE THREAD

**Issue by billalkuet07** [facebookresearch/detr#460]
## ❓ How to do something using DETR

Describe what you want to do, including:
1. what inputs you will provide, if any:
I am trying to train the DETR model with my custom dataset and the code is running. I used "python -m modified_main --coco_path /workspace --eval --batch_size 2 --epochs 150" to run my code. However, I am getting same  class_error: 100.00 and other errors fluctuating near some value. I do not know what is happening. Could you suggest me something? In modified_main.py, I just replaced the torch.distributed for my single GPU. My image size is 
2347*3129. Is this image size is an issue?

2. Some of my sample output after one epoch:
DONE (t=0.23s).
IoU metric: bbox
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.000

Epoch: [1]  [ 550/1999]  eta: 0:05:32  lr: 0.000100  **class_error: 100.00**  loss: 19.5869 (19.1807)  loss_ce: 0.6795 (0.7664)  loss_bbox: 1.1938 (1.2282)  loss_giou: 1.2004 (1.2446)  loss_ce_0: 0.6931 (0.7773)  loss_bbox_0: 1.2036 (1.2411)  loss_giou_0: 1.2390 (1.2299)  loss_ce_1: 0.7259 (0.7792)  loss_bbox_1: 1.2444 (1.1910)  loss_giou_1: 1.1770 (1.2021)  loss_ce_2: 0.6822 (0.7717)  loss_bbox_2: 1.1273 (1.1854)  loss_giou_2: 1.1401 (1.2035)  loss_ce_3: 0.6829 (0.7605)  loss_bbox_3: 1.2932 (1.2063)  loss_giou_3: 1.3359 (1.2107)  loss_ce_4: 0.6965 (0.7659)  loss_bbox_4: 1.3098 (1.2085)  loss_giou_4: 1.2878 (1.2084)  loss_ce_unscaled: 0.6795 (0.7664)  class_error_unscaled: 100.0000 (100.0000)  loss_bbox_unscaled: 0.2388 (0.2456)  loss_giou_unscaled: 0.6002 (0.6223)  cardinality_error_unscaled: 2.0000 (2.7069)  loss_ce_0_unscaled: 0.6931 (0.7773)  loss_bbox_0_unscaled: 0.2407 (0.2482)  loss_giou_0_unscaled: 0.6195 (0.6150)  cardinality_error_0_unscaled: 2.0000 (2.6289)  loss_ce_1_unscaled: 0.7259 (0.7792)  loss_bbox_1_unscaled: 0.2489 (0.2382)  loss_giou_1_unscaled: 0.5885 (0.6011)  cardinality_error_1_unscaled: 2.0000 (2.6216)  loss_ce_2_unscaled: 0.6822 (0.7717)  loss_bbox_2_unscaled: 0.2255 (0.2371)  loss_giou_2_unscaled: 0.5701 (0.6017)  cardinality_error_2_unscaled: 2.0000 (2.7060)  loss_ce_3_unscaled: 0.6829 (0.7605)  loss_bbox_3_unscaled: 0.2586 (0.2413)  loss_giou_3_unscaled: 0.6679 (0.6054)  cardinality_error_3_unscaled: 2.0000 (2.7078)  loss_ce_4_unscaled: 0.6965 (0.7659)  loss_bbox_4_unscaled: 0.2620 (0.2417)  loss_giou_4_unscaled: 0.6439 (0.6042)  cardinality_error_4_unscaled: 2.0000 (2.6969)  time: 0.1999  data: 0.0099  max mem: 3879

Epoch: [1]  [ 560/1999]  eta: 0:05:29  lr: 0.000100  **class_error: 100.00**  loss: 19.9208 (19.1622)  loss_ce: 0.6487 (0.7652)  loss_bbox: 1.1147 (1.2241)  loss_giou: 1.2055 (1.2434)  loss_ce_0: 0.6673 (0.7761)  loss_bbox_0: 1.0820 (1.2378)  loss_giou_0: 1.1724 (1.2275)  loss_ce_1: 0.7475 (0.7791)  loss_bbox_1: 1.2968 (1.1942)  loss_giou_1: 1.2224 (1.2024)  loss_ce_2: 0.6801 (0.7703)  loss_bbox_2: 1.1346 (1.1877)  loss_giou_2: 1.3087 (1.2055)  loss_ce_3: 0.6796 (0.7587)  loss_bbox_3: 1.1990 (1.2035)  loss_giou_3: 1.3358 (1.2101)  loss_ce_4: 0.6965 (0.7645)  loss_bbox_4: 1.0494 (1.2046)  loss_giou_4: 1.1900 (1.2073)  loss_ce_unscaled: 0.6487 (0.7652)  class_error_unscaled: 100.0000 (100.0000)  loss_bbox_unscaled: 0.2229 (0.2448)  loss_giou_unscaled: 0.6027 (0.6217)  cardinality_error_unscaled: 2.0000 (2.7014)  loss_ce_0_unscaled: 0.6673 (0.7761)  loss_bbox_0_unscaled: 0.2164 (0.2476)  loss_giou_0_unscaled: 0.5862 (0.6138)  cardinality_error_0_unscaled: 2.0000 (2.6221)  loss_ce_1_unscaled: 0.7475 (0.7791)  loss_bbox_1_unscaled: 0.2594 (0.2388)  loss_giou_1_unscaled: 0.6112 (0.6012)  cardinality_error_1_unscaled: 1.5000 (2.6034)  loss_ce_2_unscaled: 0.6801 (0.7703)  loss_bbox_2_unscaled: 0.2269 (0.2375)  loss_giou_2_unscaled: 0.6544 (0.6028)  cardinality_error_2_unscaled: 2.0000 (2.7005)  loss_ce_3_unscaled: 0.6796 (0.7587)  loss_bbox_3_unscaled: 0.2398 (0.2407)  loss_giou_3_unscaled: 0.6679 (0.6050)  cardinality_error_3_unscaled: 2.0000 (2.7023)  loss_ce_4_unscaled: 0.6965 (0.7645)  loss_bbox_4_unscaled: 0.2099 (0.2409)  loss_giou_4_unscaled: 0.5950 (0.6036)  cardinality_error_4_unscaled: 2.0000 (2.6916)  time: 0.2260  data: 0.0096  max mem: 3879

NOTE:

1. Only general answers are provided.
   If you want to ask about "why X did not work", please use the
   [Unexpected behaviors](https://github.com/facebookresearch/detr/issues/new/choose) issue template.

2. About how to implement new models / new dataloader / new training logic, etc., check documentation first.

3. We do not answer general machine learning / computer vision questions that are not specific to DETR, such as how a model works, how to improve your training/make it converge, or what algorithm/methods can be used to achieve X.

**Comment by SV1367** [comment]
I am having the same behaviour. I am training the original Coco 2017 dataset
Epoch: [0]  [  120/59143]  eta: 3:16:06  lr: 0.000100  class_error: 77.78  loss: 49.6219 (51.0705)  loss_ce: 1.8552 (2.1018)  loss_bbox: 3.7616 (3.9300)  loss_giou: 2.1210 (2.3823)  loss_ce_0: 1.9220 (2.0926)  loss_bbox_0: 3.9041 (3.9826)  loss_giou_0: 2.3004 (2.4286)  loss_ce_1: 1.9245 (2.0941)  loss_bbox_1: 4.0912 (4.0433)  loss_giou_1: 2.4666 (2.4312)  loss_ce_2: 1.9347 (2.1058)  loss_bbox_2: 4.0718 (4.0280)  loss_giou_2: 2.4875 (2.4332)  loss_ce_3: 1.9529 (2.1137)  loss_bbox_3: 4.0379 (4.0203)  loss_giou_3: 2.4030 (2.4352)  loss_ce_4: 1.9162 (2.1087)  loss_bbox_4: 3.8730 (3.9408)  loss_giou_4: 2.2969 (2.3982)  loss_ce_unscaled: 1.8552 (2.1018)  class_error_unscaled: 100.0000 (99.7770)  loss_bbox_unscaled: 0.7523 (0.7860)  loss_giou_unscaled: 1.0605 (1.1912)  cardinality_error_unscaled: 5.0000 (7.1942)  loss_ce_0_unscaled: 1.9220 (2.0926)  loss_bbox_0_unscaled: 0.7808 (0.7965)  loss_giou_0_unscaled: 1.1502 (1.2143)  cardinality_error_0_unscaled: 5.0000 (7.2810)  loss_ce_1_unscaled: 1.9245 (2.0941)  loss_bbox_1_unscaled: 0.8182 (0.8087)  loss_giou_1_unscaled: 1.2333 (1.2156)  cardinality_error_1_unscaled: 5.0000 (7.2769)  loss_ce_2_unscaled: 1.9347 (2.1058)  loss_bbox_2_unscaled: 0.8144 (0.8056)  loss_giou_2_unscaled: 1.2437 (1.2166)  cardinality_error_2_unscaled: 5.0000 (7.2893)  loss_ce_3_unscaled: 1.9529 (2.1137)  loss_bbox_3_unscaled: 0.8076 (0.8041)  loss_giou_3_unscaled: 1.2015 (1.2176)  cardinality_error_3_unscaled: 5.0000 (7.2810)  loss_ce_4_unscaled: 1.9162 (2.1087)  loss_bbox_4_unscaled: 0.7746 (0.7882)  loss_giou_4_unscaled: 1.1484 (1.1991)  cardinality_error_4_unscaled: 5.0000 (7.3058)  time: 0.2021  data: 0.0043  max mem: 4400

**Comment by D10752002** [comment]
@billalkuet07 @SV1367 
Did you find and fixes for this constant class error?? 
I'm training on custom dataset too.

**Comment by SV1367** [comment]
Sorry. no joy and no time.... :-(
